apps/system/test/unit/task_manager_test.js

```diff
@@ -242,6 +242,22 @@ suite('system/TaskManager >', function() {
       clock.tick(TICK_SHOW_HIDE_MS);
     });
 
+    test('_isTransitioning should be true during waitForAppToClose', (done) => {
+      tm.hide().then(() => {
+        var waitForAppToClose = TaskManagerUtils.waitForAppToClose;
+        TaskManagerUtils.waitForAppToClose = function() {
+          TaskManagerUtils.waitForAppToClose = waitForAppToClose;
+          assert.isTrue(tm._isTransitioning);
+          return Promise.resolve();
+        };
+
+        var show = tm.show();
+        clock.tick(TICK_SHOW_HIDE_MS);
+        return show;
+      }).then(done, done);
+      clock.tick(TICK_SHOW_HIDE_MS);
+    });
```

Comment:
can we make the test more "external" by checking the returned value of `respondToHierarchy` instead of inspecting the `_isTransitioning` flag?